code/datums/elements/immerse.dm

```diff
@@ -0,0 +1,387 @@
+/**
+ * A visual element that makes movables entering the attached turfs look immersed into that turf.
+ *
+ * Abandon all hope, ye who read forth, for this immerse works on mind-numbing workarounds,
+ */
+/datum/element/immerse
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY | ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	///An association list of turfs that have this element attached and their affected contents.
+	var/list/attached_turfs_and_movables = list()
+
+	/**
+	 * A list of movables that shouldn't be affected by the element, either because it'd look bad
+	 * or barely perceptible.
+	 */
+	var/static/list/movables_to_ignore
+	///A list of icons generated from a target and a mask, later used as appearances for the overlays.
+	var/static/list/generated_immerse_icons = list()
+	///A list of instances of /atom/movable/immerse_overlay then used as visual overlays for the immersed movables.
+	var/list/generated_visual_overlays = list()
+	///An association list of movables as key and overlays as assoc.
+	var/list/immersed_movables
+
+	var/icon
+	var/icon_state
+	var/mask_icon
+	var/color
+	var/alpha
+
+/datum/element/immerse/Attach(turf/target, icon, icon_state, mask_icon, color = "#777777", alpha = 180)
+	. = ..()
+	if(!isturf(target) || !icon || !icon_state || !mask_icon)
+		return ELEMENT_INCOMPATIBLE
+
+	if(!movables_to_ignore)
+		movables_to_ignore = typecacheof(list(
+			/obj/effect,
+			/mob/dead,
+			/obj/projectile,
+		))
+
+		movables_to_ignore += GLOB.WALLITEMS_INTERIOR
+		movables_to_ignore += GLOB.WALLITEMS_EXTERIOR
+
+	src.icon = icon
+	src.icon_state = icon_state
+	src.color = color
+	src.alpha = alpha
+	src.mask_icon = mask_icon
+
+	/**
+	 * Hello, you may be wondering why we're blending icons and not simply
+	 * overlaying one mutable appearance with the blend multiply on another.
+	 * Well, the latter option doesn't work as neatly when ultimately added
+	 * to an atom with the KEEP_TOGETHER appearance flag, with the mask also
+	 * showing on said atom, while we don't want it to.
+	 */
+	var/icon/immerse_icon = generated_immerse_icons["[icon]-[icon_state]-[mask_icon]"]
+	if(!immerse_icon)
+		immerse_icon = icon(icon, icon_state)
+		var/icon/sub_mask = icon('icons/effects/effects.dmi', mask_icon)
+		immerse_icon.Blend(sub_mask, ICON_MULTIPLY)
+		immerse_icon = fcopy_rsc(immerse_icon)
+		generated_immerse_icons["[icon]-[icon_state]-[mask_icon]"] = immerse_icon
```

Comment:
can't you use KEEP_APART for this? or do we want to be kept together for some reason or another

Reply:
Ummmmh... it would work, alas using KEEP_APART caused overlay to also inset into the icons of other movables on the same plane last I checked. 😭